src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1963,7 +1963,46 @@ public TryWriteInterpolatedStringHandler(int literalLength, int formattedCount,
             /// <summary>Writes the specified string to the handler.</summary>
             /// <param name="value">The string to write.</param>
             /// <returns>true if the value could be formatted to the span; otherwise, false.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public bool AppendLiteral(string value)
+            {
+                // See comment on inlining and special-casing in DefaultInterpolatedStringHandler.AppendLiteral.
+
+                if (value.Length == 1)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < (uint)destination.Length)
+                    {
+                        destination[pos] = value[0];
+                        _pos = pos + 1;
+                        return true;
+                    }
+
+                    return Fail();
+                }
+
+                if (value.Length == 2)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)(pos + 1) < (uint)destination.Length)
+                    {
+                        Unsafe.As<char, int>(ref MemoryMarshal.GetReference(destination)) = Unsafe.As<char, int>(ref value.GetRawStringData());
```

Comment:
This could result in the generation of misaligned ref ints and should be using `Unsafe.WriteUnaligned` and `Unsafe.ReadUnaligned`.

(Same comment for other occurrence of pattern.)

Reply:
Ok.  I'll need to use AsBytes as well.